plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineLibrariesSelectorGroup.java

```diff
@@ -107,55 +104,209 @@ private static String getLibraryName(Library library) {
+     */
+    @Override
+    public Boolean get() {
+      return libraryButton.getData(BUTTON_MANUAL_SELECTION_KEY) != null;
     }
   }
 
-  private Button getButtonForLibraryId(String libraryId) {
-    for (Button button : libraryButtons) {
-      Library library = (Library) button.getData();
-      if (library.getId().equals(libraryId)) {
-        return button;
+  /**
+   * Tracks if the checkbox has been explicitly clicked by the user.
+   */
+  private static final class ManualSelectionTracker implements SelectionListener {
+    @Override
+    public void widgetSelected(SelectionEvent event) {
+      setManualSelection(event);
+    }
+
+    @Override
+    public void widgetDefaultSelected(SelectionEvent event) {
+      setManualSelection(event);
+    }
+
+    private void setManualSelection(SelectionEvent event) {
+      Button source = (Button) event.getSource();
+      if (event.getSource() instanceof Button && (source.getStyle() & SWT.CHECK) != 0) {
+        Button button = source;
+        button.setData(BUTTON_MANUAL_SELECTION_KEY, button.getSelection() ? new Object() : null);
       }
     }
-    return null;
   }
 
-  private Realm getDisplayRealm() {
-    return DisplayRealm.getRealm(parentDialog.getControl().getDisplay()); 
+  /**
+   * Returns a computed value based on whether the associated library is in a list or not. 
+   */
+  private static final class DependentLibrarySelected extends ComputedValue {
+    private String libraryId;
+    private IObservableList libraries;
+    private Getter<Boolean> condition;
+    private boolean selectedResult;
+
+    /**
+     * @param libraries the list of libraries to search
+     * @param libraryId the id of the library to be searched for
+     * @param resultIfFound value returned by {@link #calculate()} if the library is found
+     */
+    private DependentLibrarySelected(Realm realm,
+                                     IObservableList libraries,
+                                     String libraryId,
+                                     final boolean resultIfFound) {
+      this(realm, libraries, libraryId, resultIfFound, new Getter<Boolean>() {
+        @Override
+        public Boolean get() {
+          return !resultIfFound;
+        }});
+    }
+
+    /**
+     * @param libraries the list of libraries to search
+     * @param libraryId the id of the library to be searched for
+     * @param resultIfFound value returned by {@link #calculate()} if the library is found
+     * @param resultIfNotFound if the library is not found in the list, return the result of <code>condition.get()</code>
+     */
+    private DependentLibrarySelected(Realm realm,
+                                     IObservableList libraries,
+                                     String libraryId,
+                                     boolean resultIfFound,
+                                     Getter<Boolean> resultIfNotFound) {
```

Comment:
Probably the naming is wrong, but the intent here is that if result not found we give a chance to clients to define the result instead of defaulting to `!resultIfFound`